addons/mail/static/src/messaging/entity/thread_viewer/thread_viewer.js

```diff
@@ -191,14 +221,44 @@ function ThreadViewerFactory({ Entity }) {
             });
 
             if (thread && this.thread !== thread) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this.link({ thread });
                 if (!this.threadCache.isLoaded && !this.threadCache.isLoading) {
                     this.threadCache.loadMessages();
                 }
             }
             if (this.threadCache !== prevThreadCache) {
+                if (this._loaderTimeout) {
+                    clearTimeout(this._loaderTimeout);
+                }
                 this._writeComponentHint('change-of-thread-cache');
             }
+
+            if(!this.threadCache) {
+                this._write({
+                    hasLoader:false,
+                    _loaderTimeout: null
```

Comment:
You can move attribute of timeout away from `_write`. The purpose of this method is to make such attributes observable from component, whether this comes from entity instances or classes. Classes are not observed due to being functions, which is why there are some tricks with `static _write()`.

Here it doesn't matter whether it is observed or not.

Reply:
so :

```
this._write({ hasLoader:false });
this._loaderTimeout = null;
```

Reply:
well, assignment to timeout should be next to `clearTimeout()`